x/ibc-transfer/keeper/relay.go

```diff
@@ -150,51 +137,42 @@ func (k Keeper) createOutgoingPacket(
 
 func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data types.FungibleTokenPacketData) error {
 	// NOTE: packet data type already checked in handler.go
-
-	if len(data.Amount) != 1 {
-		return sdkerrors.Wrapf(types.ErrOnlyOneDenomAllowed, "%d denoms included", len(data.Amount))
-	}
-
-	prefix := types.GetDenomPrefix(packet.GetDestPort(), packet.GetDestChannel())
-	source := strings.HasPrefix(data.Amount[0].Denom, prefix)
+	amount := data.Amount
 
 	// decode the receiver address
 	receiver, err := sdk.AccAddressFromBech32(data.Receiver)
 	if err != nil {
 		return err
 	}
 
-	if source {
+	if data.Source {
 
 		// mint new tokens if the source of the transfer is the same chain
 		if err := k.bankKeeper.MintCoins(
-			ctx, types.ModuleName, data.Amount,
+			ctx, types.ModuleName, sdk.NewCoins(amount),
```

Comment:
I believe this should still check if `amount` is prefixed as expected before minting

Reply:
Preferably the prefixing logic happens here, rather than in SendPacket

Reply:
makes sense to me to check the prefix here, @cwgoes should this be added to the spec?

Reply:
yes, thanks - https://github.com/cosmos/ics/commit/4509373f3b46c161b648338bc3d220c7b937a87d